pallets/creators-staking/src/lib.rs

```diff
@@ -0,0 +1,878 @@
+            space_id: SpaceId,
+            // #[pallet::compact]
+            // amount: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            // Creator must exist and it has to be unregistered
+            let creator_info =
+                RegisteredCreators::<T>::get(space_id).ok_or(Error::<T>::NotOperatedCreator)?;
+
+            let unregistered_era = if let CreatorState::Unregistered(x) = creator_info.state {
+                x
+            } else {
+                return Err(Error::<T>::NotUnregisteredCreator.into());
+            };
+
+            // There should be some leftover staked amount
+            let mut staker_info = Self::staker_info(&staker, space_id);
+            let staked_value = staker_info.latest_staked_value();
+            ensure!(staked_value > Zero::zero(), Error::<T>::NotStakedCreator);
+
+            // Don't allow withdrawal until all rewards have been claimed.
+            let (claimable_era, _) = staker_info.claim();
+            ensure!(
+                claimable_era >= unregistered_era || claimable_era.is_zero(),
+                Error::<T>::UnclaimedRewardsRemaining
+            );
+
+            // Unlock the staked amount immediately. No unbonding period for this scenario.
+            let mut ledger = Self::ledger(&staker);
+            ledger.locked = ledger.locked.saturating_sub(staked_value);
+            Self::update_ledger(&staker, ledger);
+
+            Self::update_staker_info(&staker, space_id, Default::default());
+
+            let current_era = Self::current_era();
+            GeneralEraInfo::<T>::mutate(&current_era, |value| {
+                if let Some(x) = value {
+                    x.staked = x.staked.saturating_sub(staked_value);
+                    x.locked = x.locked.saturating_sub(staked_value);
+                }
+            });
+
+            Self::deposit_event(Event::<T>::WithdrawnFromUnregistered {
+                who: staker,
+                amount: staked_value,
+            });
+
+            Ok(().into())
+        }
+
+        // Not sure here whether to calculate total rewards for all creators
+        //  or to withdraw per-creator rewards (preferably)
+        #[pallet::call_index(7)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        pub fn claim_staker_reward(origin: OriginFor<T>, space_id: SpaceId, restake: bool) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            // Ensure we have something to claim
+            let mut staker_info = Self::staker_info(&staker, space_id);
+            let (era, staked) = staker_info.claim();
+            ensure!(staked > Zero::zero(), Error::<T>::NotStakedCreator);
+
+            let creator_info =
+                RegisteredCreators::<T>::get(space_id).ok_or(Error::<T>::NotOperatedCreator)?;
```

Comment:
The phrases "NonRegisteredCreator" and "NotRegisteredCreator" are similar, and both indicate that the creator is not registered in the system. However, there is a slight semantic difference between them:

1. NonRegisteredCreator: The prefix "non-" means "not" in a general sense, and it usually denotes the absence of a quality or state. "NonRegisteredCreator" might be read as a category or class of creators that have never been registered.
2. NotRegisteredCreator: The word "not" is more specific, and it denotes a negation of a state or condition. "NotRegisteredCreator" would imply that the creator is expected to be registered, but the expected registration is not present. This might be more suitable if the error is being raised in a context where registration is normally expected.

In the context of error naming for a creator not being registered, "**NotRegisteredCreator**" may be more fitting as it directly states that the expected registration state for this particular creator is not fulfilled.